lib/syntax_tree/rake/task.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+require "rake"
+require "rake/tasklib"
+
+module SyntaxTree
+  module Rake
+    # A Rake task that runs check and format on a set of source files.
+    #
+    # Example:
+    #
+    #   require 'syntax_tree/rake/task'
+    #
+    #   SyntaxTree::Rake::Task.new do |t|
+    #     t.source_files = '{app,config,lib}/**/*.rb'
+    #   end
+    #
+    # This will create task that can be run with:
+    #
+    #   rake syntax_tree:check_and_format
+    class Task < ::Rake::TaskLib
+      # Glob pattern to match source files.
+      # Defaults to 'lib/**/*.rb'.
+      attr_accessor :source_files
+
+      def initialize
+        @source_files = "lib/**/*.rb"
+
+        yield self if block_given?
+        define_task
+      end
+
+      private
+
+      def define_task
+        desc "Runs syntax_tree over source files"
+        task(:check_and_format) { run_task }
+      end
+
+      def run_task
+        %w[check format].each do |command|
+          SyntaxTree::CLI.run([command, source_files].compact)
+        end
+
+        # TODO: figure this out
+        # exit($?.exitstatus) if $?&.exited?
```

Comment:
I'm still investigating if that's something I need for this task.

Reply:
@kddnewton Do you recall why we need this explicit `exit()` call? This line force the test suite to suddenly finish with `with status (1)` even though debugging I can see that `$?.exitstatus == 0` and `$?&.exited? == true`

Reply:
Yeah that was just because prettier was running as a separate process so if it exited prematurely then we wanted to exit with the same exit code. It shouldn't be necessary here.